sycl/test-e2e/Graph/graph_common.hpp

```diff
@@ -14,6 +14,16 @@ namespace exp_ext = sycl::ext::oneapi::experimental;
 // Make tests less verbose by using sycl namespace.
 using namespace sycl;
 
+// Macro for wrapping depends on calls when add_node is used so they are not
+// used in the explicit API
+#ifdef GRAPH_E2E_RECORD_REPLAY
+#define DEPENDS_ON(CGH, EVENTS) CGH->depends_on(EVENTS)
+#else
+#define DEPENDS_ON(CGH, EVENTS)                                                \
+  do {                                                                         \
+  } while (0)
```

Comment:
As an update to this I've ended up replacing this macro with a template function since handling more than one event was problematic, but it has the same function and in explicit mode the calls will still be a no-op.